BHoM_Engine/Query/Geometry.cs

```diff
@@ -94,7 +94,11 @@ private static IGeometry Geometry(this object obj)
 
         private static IGeometry Geometry(this IBHoMObject obj)
         {
-            return Reflection.Compute.RunExtensionMethod(obj, "Geometry") as IGeometry;
+            System.Reflection.MethodInfo mi = Reflection.Query.ExtensionMethodToCall(obj, "Geometry");
+            if (mi != null)
+                return Reflection.Compute.RunExtensionMethod(obj, "Geometry") as IGeometry;
+            else
+                return null;
```

Comment:
Even if this is already merged, it is worth noting that `Reflection.Compute.RunExtensionMethod(obj, mi)` should be called instead to avoid calling `ExtensionMethodToCall(obj, "Geometry")` twice

Not a massive deal breaker but something to keep in mind if this pattern is to be repeated elsewhere.

I also agree with Al that a `TryRunExtensionMethod` would keep things tidier.

Reply:
You're right @adecler , I actually thought I did that - will correct in separate PR.

Happy to push the TryRunExtensionMethod.